Move Cron package usage to native web ext Alarm API

tctara opened the pull request:
Hi @poltak,
Mind if I borrow you for a few minutes to review these changes? 
I'm trying to get  #329 done

poltak commented:
Awesome work here! Code looks great and really nice that you managed to port it over without needing to change much at all.

I would maybe put a comment on the `Math.random()` call to very briefly explain the purpose, and how it will recalc every time the BG script loads. You might as well `yarn remove cron` if all the uses of that package have been ported over now as well.

I will have a play around with the code later to make sure it's working as expected 👍

poltak commented on src/analytics/background/periodic.js:
```diff
@@ -88,18 +87,17 @@ const attemptPeriodicPing = (
-    createPeriodicEventJob('day'),
-]
+browser.alarms.onAlarm.addListener(alarm =>
+    attemptPeriodicPing(alarm.name, getActivePeriodVars(alarm.name)),
```
Just tried it out and it mostly works but you'll need to update this function's signature. Currently it is a higher-order fn of type `(period: string, activePeriodVars: any) => () => Promise<void>` but now as you're using it inside the `addListener` callback it will need to change to `(period: string, activePeriodVars: any) => Promise<void>`, or else all that inner logic won't be invoked (only need to change lines 46-49)

tctara commented:
I have:
-  Changed the function's signature
-  Remove Cron 
-  Added comment for Math.random
-  Amended the commit 

Hope everything is good for merge.

poltak commented:
Looks great and seems to work fine now. Thanks a lot @tctara!

There a similar web ext API refactoring issue in #288, if you liked this one. Just putting it out there :)
